### ISSUE TITLE
Can't migrate database - Object of class stdClass could not be converted to string

### ISSUE THREAD

**Issue by dude22072** [alexjustesen/speedtest-tracker#1168]
**Describe the bug**
Attempting to Migrate the database results in an error.

**To Reproduce**
1. Click Migrate
2. Click "Yes, migrate it"
3. Error

**Expected behavior**
Successful migration

**Environment (please complete the following information):**
 - Unraid Docker (LinuxServer.io)
 - MariaDB

**Logs**
```
[2024-02-19 23:14:36] production.ERROR: Error: Object of class stdClass could not be converted to string in /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:723
Stack trace:
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(723): PDOStatement->bindValue()
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(583): Illuminate\Database\Connection->bindValues()
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(816): Illuminate\Database\Connection->Illuminate\Database\{closure}()
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(783): Illuminate\Database\Connection->runQueryCallback()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(576): Illuminate\Database\Connection->run()
#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php(540): Illuminate\Database\Connection->statement()
#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(3336): Illuminate\Database\Connection->insert()
#7 /var/www/html/app/Actions/MigrateBadJsonResults.php(80): Illuminate\Database\Query\Builder->insert()
#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Concerns/BuildsQueries.php(174): App\Actions\MigrateBadJsonResults->App\Actions\{closure}()
#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Database/Concerns/BuildsQueries.php(116): Illuminate\Database\Query\Builder->orderedChunkById()
#10 /var/www/html/app/Actions/MigrateBadJsonResults.php(62): Illuminate\Database\Query\Builder->chunkById()
#11 [internal function]: App\Actions\MigrateBadJsonResults->handle()
#12 /var/www/html/vendor/lorisleiva/laravel-actions/src/Concerns/DecorateActions.php(39): call_user_func_array()
#13 /var/www/html/vendor/lorisleiva/laravel-actions/src/Decorators/JobDecorator.php(61): Lorisleiva\Actions\Decorators\JobDecorator->callMethod()
#14 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Lorisleiva\Actions\Decorators\JobDecorator->handle()
#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#19 /var/www/html/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(128): Illuminate\Container\Container->call()
#20 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Bus\Dispatcher->Illuminate\Bus\{closure}()
#21 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#22 /var/www/html/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(132): Illuminate\Pipeline\Pipeline->then()
#23 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(123): Illuminate\Bus\Dispatcher->dispatchNow()
#24 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Queue\CallQueuedHandler->Illuminate\Queue\{closure}()
#25 /var/www/html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#26 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(122): Illuminate\Pipeline\Pipeline->then()
#27 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(70): Illuminate\Queue\CallQueuedHandler->dispatchThroughMiddleware()
#28 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(102): Illuminate\Queue\CallQueuedHandler->call()
#29 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(439): Illuminate\Queue\Jobs\Job->fire()
#30 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(389): Illuminate\Queue\Worker->process()
#31 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(176): Illuminate\Queue\Worker->runJob()
#32 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(137): Illuminate\Queue\Worker->daemon()
#33 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(120): Illuminate\Queue\Console\WorkCommand->runWorker()
#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\Queue\Console\WorkCommand->handle()
#35 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#36 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#37 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#38 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#39 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\Container\Container->call()
#40 /var/www/html/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute()
#41 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(180): Symfony\Component\Console\Command\Command->run()
#42 /var/www/html/vendor/symfony/console/Application.php(1096): Illuminate\Console\Command->run()
#43 /var/www/html/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand()
#44 /var/www/html/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun()
#45 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#46 /var/www/html/artisan(35): Illuminate\Foundation\Console\Kernel->handle()
#47 {main}  
```

**Additional context**

**Comment by alexjustesen** [comment]
Any chance you can dig into your database and take a look at the `results_bad_json` table?

Specifically do a quick skim for records that seem out of place and don't match the bulk of the rest? I'm thinking there is a format or data set that I'm not taking into account in the code.

**Comment by dude22072** [comment]
Ah, this is entirely my error. At one point in the past I had migrated from SQLite over to MariaDB and it seems that however I went about that the first few hundred results didn't get their `data` column properly escaped (didn't have quotations at the beginning and end), so the stacktrace is probably a result of invalid JSON data not getting turned into a string correctly. I don't think this is an issue a normal user should run in to.
